.github/workflows/spec.yml

```diff
@@ -0,0 +1,38 @@
+name: "Static & Spec Tests"
+
+on: [pull_request, workflow_dispatch]
```

Comment:
@jpogran , @da-ar Thoughts on this being executable manually? Do we want to restrict to PRs only? I included `workflow_dispatch` as I thought it might be handy for us to trigger the job manually if we ever wanted to.

Reply:
my two cents is no, no manual execution. i can't imagine the scenario at the moment